src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -617,6 +625,80 @@ async Task<bool> OnPause(SessionId sessionId, JObject args, CancellationToken to
             return true;
         }
 
+        private MethodInfo LoadSymbolsOnDemand(AssemblyInfo asm, uint method_token, SessionId sessionId, CancellationToken token)
+        {
+            MethodInfo method = null;
+            var context = GetContext(sessionId);
+            ImageDebugHeader header = asm.Image.GetDebugHeader();
+            for (var i = 0; i < header.Entries.Length; i++)
+            {
+                var entry = header.Entries[i];
+                if (entry.Directory.Type == ImageDebugType.CodeView)
+                {
+                    var data = entry.Data;
+
+                    if (data.Length < 24)
+                        return null;
+
+                    var magic = (data[0]
+                        | (data[1] << 8)
+                        | (data[2] << 16)
+                        | (data[3] << 24));
+
+                    if (magic != 0x53445352)
+                        return null;
+
+                    var buffer = new byte[16];
+                    Buffer.BlockCopy(data, 4, buffer, 0, 16);
+
+                    var pdbAge = (data[20]
+                        | (data[21] << 8)
```

Comment:
Give this check and the magic check symbolic meaning or at the very least comments